utility/src/test/java/org/oppia/android/util/networking/DebugNetworkConnectionUtilTest.kt

```diff
@@ -0,0 +1,278 @@
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.networking.NetworkConnectionUtil.ConnectionStatus
+import org.oppia.android.util.utility.NetworkConnectionTestUtil
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [DebugNetworkConnectionUtil]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class DebugNetworkConnectionUtilTest {
+
+  private val NO_CONNECTION = -1
+
+  @Inject
+  lateinit var debugNetworkConnectionUtil: DebugNetworkConnectionUtil
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerDebugNetworkConnectionUtilTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeWifi_noForcedConnection_returnsWifi() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.CONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.LOCAL
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeWifi_forceCellularConnection_returnsCellular() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.CELLULAR)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeWifi_forceNoConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.NONE)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveWifi_noForcedConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveWifi_forceWifiConnection_returnsNone() {
```

Comment:
This scenario doesn't make sense to me--shouldn't it return wifi since that's being forced? I suspect this is related to some of my earlier comments.

Reply:
Test corrected. PTAL.